grpc/src/test/java/com/linecorp/armeria/server/grpc/UnframedGrpcServiceTest.java

```diff
@@ -139,4 +140,17 @@ private static UnframedGrpcService buildUnframedGrpcService(BindableService bind
                                                 .enableUnframedRequests(true)
                                                 .build();
     }
+
+    @Test
+    void shouldThrowExceptionIfUnframedRequestHandlerAddedButUnframedRequestsAreDisabled() {
+        IllegalStateException exception = assertThrows(IllegalStateException.class, () ->
+                GrpcService.builder()
+                           .setMaxInboundMessageSizeBytes(MAX_MESSAGE_BYTES)
+                           .setMaxOutboundMessageSizeBytes(MAX_MESSAGE_BYTES)
+                           .supportedSerializationFormats(GrpcSerializationFormats.values())
+                           .enableUnframedRequests(false)
+                           .unframedGrpcErrorHandler(UnframedGrpcErrorHandler.of())
+                           .build());
+        assertThat(exception).hasMessage("'unframedGrpcErrorHandler' can only be set if unframed requests are enabled");
+    }
```

Comment:
Could you add integration tests that a client receives either JSON or plain text depending on the `UnframedGrpcErrorHandler`?

Reply:
Can you point me to any existing integration tests if any?